run_inference.py

```diff
@@ -67,13 +67,16 @@ def main():
 
         output = disp_net(tensor_img)[0]
 
+        file_ = file.replace(args.dataset_dir, '')
+        file_ = '-'.join(file_.split('/'))
+                                     
         if args.output_disp:
             disp = (255*tensor2array(output, max_value=None, colormap='bone', channel_first=False)).astype(np.uint8)
-            imsave(output_dir/'{}_disp{}'.format(file.namebase,file.ext), disp)
+            imsave(output_dir/'{}_disp{}'.format(file_,file.ext), np.transpose(disp, (1,2,0)))
```

Comment:
Be careful, here you are saveing the extension of the file within the path to get something like
`output_dir/img.jpg-disp.jpg` which is not very nice.
refer to my first comment in which we split the path beforhand with splitext.

the resulting line will be
```python
imsave(output_dir/'{}_depth{}'.format(file_name,file_ext), np.transpose(depth, (1,2,0)))
```

Reply:
Sorry this is a screw-up on my end, I meant ```imsave(output/'{}_disp{}'.format(file_), np.transpose(1,2,0))```, but your way is indeed better!

Reply:
This is not going to work, the format function expects 2 arguments.
Namely, we want to go from `img.jpg` to `img_depth.jpg` so we have to decompose the filename into namebase and extension.